### ISSUE TITLE
Not works in this case:

### ISSUE THREAD

**Issue by Karaulov** [TsudaKageyu/minhook#52]
http://i.imgur.com/5yXJfKC.png
![5yxjfkc](https://user-images.githubusercontent.com/6445336/28502524-d3bf7c22-6ffc-11e7-9b84-d7e38e53b8e2.png)

**Comment by Karaulov** [comment]
Can be fixed?

**Comment by RaMMicHaeL** [comment]
What's the issue? What error do you get?

**Comment by Karaulov** [comment]
I don't know. At this moment i use another hooked function.

Problem in hook, hook just can not be set at this address. It is possible ?

At left side: function prototype and pseudocode, at right function asm disassembled output.

**Comment by RaMMicHaeL** [comment]
What is the error code that the `MH_CreateHook` function returns?

**Comment by Karaulov** [comment]
Sorry i don't know, at this moment i don't use this hook. If need i can try check it.

**Comment by RaMMicHaeL** [comment]
The function assembly looks OK. If you'd like to get help, you need to provide more information.

**Comment by Karaulov** [comment]
Yes. At hook you can see what minhook not install hook here. (assembly is clean)

Please wait i try to get error code.

**Comment by Karaulov** [comment]
code:


```
	ijlInit_org = ( ijlInit_p )( LoadLibrary( "ijl15.dll" ) + 0x32240 );
	auto err1 = MH_CreateHook( ijlInit_org, &ijlInit_my, reinterpret_cast< void** >( &ijlInit_ptr ) );
	auto err2 = MH_EnableHook( ijlInit_org );

	char tmperrcode[ 100 ];
	sprintf_s( tmperrcode, "MH_CreateHook = %s\nMH_EnableHook = %s", MH_StatusToString( err1 ), MH_StatusToString( err2 ) );
	MessageBox(0, tmperrcode,"",0 );

```

**MH_CreateHook = MH_ERROR_NOT_EXECUTABLE
MH_EnableHook = MH_ERROR_NOT_CREATED**

**Comment by Karaulov** [comment]
**_ijlInit_org and ijlInit_my is available._**

**Comment by RaMMicHaeL** [comment]
`MH_ERROR_NOT_EXECUTABLE` means that the pointer that you're passing is not executable code.
1. Print the addresses of `LoadLibrary` and of `ijlInit_org`. and see if the addresses are correct.
2. Upload the DLL and the demo above as an exe, I'll take a look.

**Comment by Karaulov** [comment]
Yes problem in ijlInit_org addr. I fix it and get next problem:

In hooked ijlInit_org:

![auqrseu 1](https://user-images.githubusercontent.com/6445336/30803617-ea480608-a1f2-11e7-8bd3-27aee725709e.png)

In ijlInit_my:

![xh6zf8p 1](https://user-images.githubusercontent.com/6445336/30803649-122d0a92-a1f3-11e7-9de1-c4e6eb54f812.png)

In ijlInit_ptr:

![huu8iva 1](https://user-images.githubusercontent.com/6445336/30803678-2d9cf436-a1f3-11e7-9036-a63e8beecb47.png)


org ijlInit_org

![wbll6ft 1](https://user-images.githubusercontent.com/6445336/30803840-ce45ff0e-a1f3-11e7-8e46-8e7bdbbdaa58.png)


**Minhook crash this function or all okay ? :D**

**Comment by RaMMicHaeL** [comment]
It's probably OK. The question marks in `ijlInit_ptr` is an issue that I've already seen with CheatEngine, try another debugger. `ijlInit_org` shows broken commands because the bytes after the jump are part of an overridden command, which will not be used while the hook is enabled.

**Comment by Karaulov** [comment]
Yes Cheat Engine bad debugger. (Always crashes, etc)

All works.



Minhook support multiple hooks ? (For example hook from several programs (dll) files hook same functions ?)


For exmaple:
```

MH_CreateHook(messagebox, messagebox_my, messagebox_ptr)
MH_CreateHook(messagebox, messagebox_my2, messagebox_ptr2)
MH_CreateHook(messagebox, messagebox_my3, messagebox_ptr3)

when messagebox called, by queue call messagebox_my  messagebox_my2  messagebox_my3 ?
```

Or this is impossible ?

**Comment by RaMMicHaeL** [comment]
> Minhook support multiple hooks ?

No, but I have a fork that does. See https://github.com/TsudaKageyu/minhook/issues/49#issuecomment-295342460.
